examples/provider_upgrade_test.go

```diff
@@ -0,0 +1,411 @@
+
+// Experimental provider upgrade tests.
+//
+// The objective of these tests is to make sure the provider release candidate will not generate any
+// Update or Replace plans for Pulumi stacks deployed on a previous baseline version of the
+// provider.
+//
+// Note on build flags above --^ these tests currently uses YAML only but our build workflow matrix
+// does not run YAML-only test a the moment, and it splits tests by language build tag; for this
+// reason the file is marked as another language so that the test runs only once.
+
+package examples
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	jsonpb "google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"gopkg.in/yaml.v2"
+
+	pfbridge "github.com/pulumi/pulumi-terraform-bridge/pf/tfbridge"
+	testutils "github.com/pulumi/pulumi-terraform-bridge/testing/x"
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+
+	providerRC "github.com/pulumi/pulumi-aws/provider/v6"
+	"github.com/pulumi/pulumi-aws/provider/v6/pkg/version"
+)
+
+const (
+	acceptEnvVar   = "PULUMI_ACCEPT"
+	providerBinary = "pulumi-resource-aws"
+)
+
+var (
+	accept bool = cmdutil.IsTruthy(os.Getenv(acceptEnvVar))
+)
+
+func TestProviderUpgradeQuick(t *testing.T) {
+	info := newProviderUpgradeInfo(t)
+
+	bytes, err := os.ReadFile(info.grpcFile)
+	require.NoErrorf(t, err, "No pre-recorded gRPC log found, try to run TestProviderUpgradeRecord")
+
+	eng := &mockPulumiEngine{
+		provider:              providerServer(t),
+		lastCheckRequestByURN: map[string]*pulumirpc.CheckRequest{},
+	}
+
+	for _, line := range strings.Split(string(bytes), "\n") {
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+		line = ignoreStables(t, line)
+		eng.replayGRPCLog(t, line)
+	}
+	require.NotEmptyf(t, eng.verifiedDiffResourceCounter, "Need at least one replay test")
+}
+
+func TestProviderUpgradeRecord(t *testing.T) {
+	if !accept {
+		t.Skipf("Skipping; to record baselines set %s env var to true", acceptEnvVar)
+	}
+	info := newProviderUpgradeInfo(t)
+	ambientProvider, _ := exec.LookPath(providerBinary)
+	require.Emptyf(t, ambientProvider, "please remove the provider from PATH")
+	ensureFolderExists(t, info.recordingDir)
+	deleteFileIfExists(t, info.stateFile)
+	deleteFileIfExists(t, info.grpcFile)
+	test := info.opts.With(integration.ProgramTestOptions{
+		Env: append(info.opts.Env, fmt.Sprintf("PULUMI_DEBUG_GRPC=%s", info.grpcFile)),
+		ExportStateValidator: func(t *testing.T, state []byte) {
+			writeFile(t, info.stateFile, state)
+			t.Logf("wrote %s", info.stateFile)
+		},
+
+		// TODO eks.Cluster fails refresh on 5.42.0
+		SkipRefresh: true,
+	})
+	integration.ProgramTest(t, &test)
+}
+
+func TestProviderUpgrade(t *testing.T) {
```

Comment:
`go test ---run=TestProviderUpgrade` is globbed. to match `TestProviderUpgrade`, `TestProviderUpgradeQuick` and `TestProviderUpgradeRecord`. Can we change the name so we can run just this test?
```suggestion
func TestProviderUpgradeFull(t *testing.T) {
```

Reply:
Ok this got addressed in a different way.